react-client/src/components/UserpickApp.jsx

```diff
@@ -0,0 +1,211 @@
+import React from 'react';
+import $ from 'jquery'
+
+class UserAppoinment extends React.Component{
+	constructor(props){
+		super(props);
+		this.state={
+			show:false,
+			doctors:[],
+			doctorsId:[],
+			doctorsHour:[],
+			doctor:'',
+			doctorHour:0,
+			months:["January","February","March","April","May","June","July","August","September","October","November","December"],
+			month:0,
+			days:[],
+			day:0,
+			hours:[],
+			hour:0
+		}
+		this.doctorhandleChange=this.doctorhandleChange.bind(this)
+		this.show=this.show.bind(this);
+		this.monthHandelChange=this.monthHandelChange.bind(this);
+		this.daysHandleChange=this.daysHandleChange.bind(this);
+		this.send=this.send.bind(this);
+		this.hourHandleChange=this.hourHandleChange.bind(this);
+	}
+
+	componentDidMount(){
+		var that=this;
+		var arr=[];
+		var arr1=[];
+		$.ajax({
+			type:'get',
+			url:'/Doctor/retrieve',
+			success:function(data){
+				for(var i=0;i<data.length;i++){
+					arr.push(data[i]._id);
+					arr1.push(data[i].hoursOfWork);
+				}
+				that.setState({
+					doctors:data,
+					doctorsId:arr,
+					doctorsHour:arr1
+				})
+			}
+		})	
+	}
+
+	show(){
+		this.setState({
+			show:!this.state.show
+		})
+
+	}
+
+	doctorhandleChange(e){
+		var arr2=[];
+		var x=this.state.doctorsId.indexOf(e.target.value);
```

Comment:
Make your variable names more descriptive. No one is going to know what "x" or "xx" means, which will make it harder for other people (especially your teammates) to read and understand your code.